qiskit_nature_pyscf/pyscf_ground_state_solver.py

```diff
@@ -0,0 +1,214 @@
+                f"problem of type '{type(problem)}'."
+            )
+
+        e_ints = problem.hamiltonian.electronic_integrals
+
+        restricted_spin = e_ints.beta_alpha.is_empty()
+
+        one_body: np.ndarray
+        two_body: np.ndarray
+
+        if restricted_spin:
+            one_body = np.asarray(problem.hamiltonian.electronic_integrals.alpha["+-"])
+            two_body = np.asarray(
+                to_chemist_ordering(
+                    problem.hamiltonian.electronic_integrals.alpha["++--"],
+                )
+            )
+        else:
+            one_body = np.asarray(
+                [
+                    problem.hamiltonian.electronic_integrals.alpha["+-"],
+                    problem.hamiltonian.electronic_integrals.beta["+-"],
+                ]
+            )
+            index_order = find_index_order(problem.hamiltonian.electronic_integrals.alpha["++--"])
+            two_body = np.asarray(
+                [
+                    to_chemist_ordering(
+                        problem.hamiltonian.electronic_integrals.alpha["++--"],
+                        index_order=index_order,
+                    ),
+                    to_chemist_ordering(
+                        problem.hamiltonian.electronic_integrals.alpha_beta["++--"],
+                        index_order=index_order,
+                    ),
+                    to_chemist_ordering(
+                        problem.hamiltonian.electronic_integrals.beta["++--"],
+                        index_order=index_order,
+                    ),
+                ]
+            )
+
+        energy, ci_vec = self.solver.kernel(
+            h1e=one_body,
+            eri=two_body,
+            norb=problem.num_spatial_orbitals,
+            nelec=problem.num_particles,
+        )
+
+        density: ElectronicDensity
+        if restricted_spin:
+            raw_density = self.solver.make_rdm1(
+                ci_vec, norb=problem.num_spatial_orbitals, nelec=problem.num_particles
+            )
+            density = ElectronicDensity.from_raw_integrals(raw_density)
+        else:
+            raw_density = self.solver.make_rdm1s(
+                ci_vec, norb=problem.num_spatial_orbitals, nelec=problem.num_particles
+            )
+            density = ElectronicDensity.from_raw_integrals(raw_density[0], h1_b=raw_density[1])
+
+        result = ElectronicStructureResult()
+        result.computed_energies = np.asarray([energy])
+        result.hartree_fock_energy = problem.reference_energy
+        result.extracted_transformer_energies = dict(problem.hamiltonian.constants.items())
+        result.nuclear_repulsion_energy = result.extracted_transformer_energies.pop(
+            "nuclear_repulsion_energy", None
+        )
+        result.num_particles = [sum(problem.num_particles)]
+        result.magnetization = [float("NaN")]
+        result.total_angular_momentum = [float("NaN")]
```

Comment:
The reason for injecting these `NaN` values is the limited capabilities of the `ElectronicStructureResult`.
When not setting these at all, printing the result will fail because it tries to iterate a `None` value. Making it an empty list will not allow the number of particles to be printed (because these are zipped over). Setting it to `[None]` breaks the computation of the inferred value of `.spin` based on `.total_angular_momentum`....

This will be addressed in Qiskit Nature directly in the future: https://github.com/qiskit-community/qiskit-nature/issues/1198
Once done, we can revisit this here :+1: